target/linux/ar71xx/base-files/etc/board.d/02_network

```diff
@@ -424,7 +424,8 @@ ar71xx_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth0" "1:lan:1" "2:lan:2" "3:lan:3" "4:lan:4"
 		;;
-	tl-wr842n-v2)
+	tl-wr842n-v2|\
+	tl-wr942n-v1)
 		ucidef_set_interfaces_lan_wan "eth1.1" "eth0"
 		ucidef_add_switch "switch0" \
 			"0@eth1" "1:lan:4" "2:lan:1" "3:lan:2" "4:lan:3"
```

Comment:
There is a piece of uboot Makefile which redefines GPIOs for uart to unpopulated 4 and 5.

`ifeq ($(CFG_BOARD_TYPE), ap151)
ifneq ($(CONFIG_NO_SERIAL),)
        @echo '#define UART_RX5_TX4           1'      >>include/config.h
        @echo '#define CONFIG_942RU_NO_SERIAL           1'      >>include/config.h
else
ifneq ($(filter $(PRODUCT_NAME), WR942NV1 WR942HPV1 ARCHERC59V1 ARCHERC58V1),)
        @echo '#define UART_RX15_TX14           1'      >>include/config.h
else`

There is another piece of code in board956x.c which defines GPIOs 14 and 15 as output and use it for USB1 and USB2 LEDs

`#ifdef CONFIG_942RU_NO_SERIAL
// set GPIO 14~22 as output
    ath_reg_rmw_clear(GPIO_OE_ADDRESS, (1 << 14) | (1 << 15) | (1 << 16) | (1 << 17) | \
                                       (1 << 18) | (1 << 19) | (1 << 20) | (1 << 21) | \
                                       (1 << 22));
#else
// set GPIO 16~22 as output
    ath_reg_rmw_clear(GPIO_OE_ADDRESS, (1 << 16) | (1 << 17) | (1 << 18) | (1 << 19) | \
                                       (1 << 20) | (1 << 21) | (1 << 22));
#endif
`

USB LEDs and UART Tx and Rx pins are hardwired on the board and AFAIK can not be used simultaneously. Is this information useful?